src/json/serialization.cr

```diff
@@ -200,7 +199,11 @@ module JSON
         {% end %}
 
         {% for name, value in properties %}
-          %var{name} = {% if value[:has_default] || value[:nilable] %} nil {% else %} uninitialized ::Union({{value[:type]}}) {% end %}
+          %var{name} = {% if value[:has_default] || value[:nilable] %}
+                         nil.as(typeof(nil, @{{name}}))
```

Comment:
This is a workaround, right? So we should add a comment to explain that.

Reply:
And `nil` doesn't need to be put through `typeof`. I suppose this would be a better (more direct) notation?
```suggestion
                         nil.as(Nil | typeof(@{{name}}))
```